reflekt-plugin/src/main/kotlin/io/reflekt/plugin/generation/code/generator/models/HelperClassGenerator.kt

```diff
@@ -110,25 +119,23 @@ abstract class HelperClassGenerator : ClassGenerator() {
     }
 
     protected fun generateNestedWhenBody(uses: ClassOrObjectUses, annotationFqNames: String, supertypeFqNames: String): CodeBlock {
-        val mainFunction = { o: Map.Entry<SupertypesToAnnotations, MutableList<KtClassOrObject>> ->
+        val mainFunction = { o: Map.Entry<SupertypesToAnnotations, List<String>> ->
             getWhenOption(o.key.annotations, wrappedCode(generateWhenBody(mapOf(o.key.supertypes to o.value), supertypeFqNames, toAddReturn = false)))
         }
-        return generateWhenBody(uses.toMap().asIterable(), annotationFqNames, mainFunction)
+        return generateWhenBody(uses.mapValues { (_, v) -> v.mapNotNull { it.fqName?.toString() } }.toMap().asIterable(), annotationFqNames, mainFunction)
     }
 
     protected companion object {
         const val WITH_SUPERTYPES_FUNCTION_NAME = "withSupertypes"
-        val WITH_SUPERTYPES_CLASS_NAME = WITH_SUPERTYPES_FUNCTION_NAME.capitalize()
+        val WITH_SUPERTYPES_CLASS_NAME = WITH_SUPERTYPES_FUNCTION_NAME.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }
```

Comment:
why do you need to call it.toString?

Reply:
Since titlecase return `string`, but `it` is `char`